pysingfel/detector/base.py

```diff
@@ -246,9 +270,59 @@ def get_photons(self, particle, device=None):
                 "Device option is deprecated. "
                 "Everything now runs on the GPU.")
 
-        raw_data = self.get_pattern_without_corrections(particle=particle)
+        raw_data = self.get_pattern_without_corrections(particle=particle,return_type="intensity")
         return self.add_correction_and_quantization(raw_data)
 
+
+    def maxRadius(self, particles):
```

Comment:
Suggest moving maxRadius() and distribute() functions to a seperate file: pysingfel/particlePlacement.py